appeals/web/src/server/lib/display-page-formatter.js

```diff
@@ -0,0 +1,119 @@
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfAppeals.length; i++) {
+			const shortAppealReference = appealShortReference(listOfAppeals[i].appealReference);
+			formattedLinks += `<a href='/appeals-service/appeal-details/${listOfAppeals[i].appealId}' class="govuk-link">${shortAppealReference}</a>`;
+			formattedLinks += i + 1 >= listOfAppeals.length ? '' : '<br>';
+		}
+		return `<span>${formattedLinks}</span>`;
+	}
+	return undefined;
+};
+
+/** @returns {string} */
+export const formatListOfListedBuildingNumbers = (
+	/** @type {string | any[]} */ listOfListedBuildingNumbers
+) => {
+	if (listOfListedBuildingNumbers.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfListedBuildingNumbers.length; i++) {
+			const listedBuildingNumber = listOfListedBuildingNumbers[i];
+			if ('listEntry' in listedBuildingNumber) {
+				formattedLinks += `<a href='https://historicengland.org.uk/listing/the-list/list-entry/${listedBuildingNumber.listEntry}' class="govuk-link">${listedBuildingNumber.listEntry}</a>`;
+				formattedLinks += i + 1 >= listOfListedBuildingNumbers.length ? '' : '<br>';
+			} else {
+				formattedLinks += `<a href='https://historicengland.org.uk} class="govuk-link"'>ListEntryNumber not yet in BD</a>`;
+				formattedLinks += i + 1 >= listOfListedBuildingNumbers.length ? '' : '<br>';
+			}
+		}
+		return `<span>${formattedLinks}</span>`;
+	}
+	return '<span>No listed building details</span>';
+};
+
+/** @returns {string} */
+export const formatDocumentValues = (
+	/** @type {number} */ appealId,
+	/** @type {any[] | any} */ listOfDocuments
+) => {
+	let formattedDocumentList = ``;
+	// If list of documents is in the correct format
+	if (
+		listOfDocuments !== null &&
+		typeof listOfDocuments === 'object' &&
+		'documents' in listOfDocuments
+	) {
+		listOfDocuments = listOfDocuments.documents;
+		if (Array.isArray(listOfDocuments)) {
+			for (let i = 0; i < listOfDocuments.length; i++) {
+				const document = listOfDocuments[i];
+				formattedDocumentList += `<a href='/documents/${appealId}/download/${document.id}/preview' target="'_docpreview'" class="govuk-link">${document.name}</a>`;
+				formattedDocumentList += i + 1 >= listOfDocuments.length ? '' : '<br>';
+			}
+			return formattedDocumentList;
+		}
+		return `<a href='/documents/${appealId}/download/${listOfDocuments.id}/preview' target="'_docpreview'" class="govuk-link">${listOfDocuments.name}</a>`;
+	}
+	// other display error but show anyway
+	else {
+		logger.error('Document not in correct format');
+		if (Array.isArray(listOfDocuments)) {
+			for (let i = 0; i < listOfDocuments.length; i++) {
+				logger.info(listOfDocuments[i]);
+				formattedDocumentList += `<a href='#' class="govuk-link">${listOfDocuments[i]}</a>`;
+				formattedDocumentList += i + 1 >= listOfDocuments.length ? '' : '<br>';
+			}
+			return formattedDocumentList;
+		}
+		logger.info(listOfDocuments);
+		return `<a href='#' class="govuk-link">${listOfDocuments}</a>`;
```

Comment:
I'm not sure what the expected behaviour is here, but this renders a link with innerText of `[object Object]` when testing locally